### PR TITLE
Cfinspec 317 support pods

### PR DESCRIPTION
**Pull request by sathish-progress**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: libraries/k8sobject.rb [hunk 1]
```diff
@@ -20,49 +21,60 @@ class K8sObject < K8sResourceBase
...
       @objtype = opts[:type] if opts[:type] ||= nil
       @objname = opts[:name] if opts[:name] ||= nil
       @objnamespace = opts[:namespace] if opts[:namespace] ||= nil
 
-      if !@objnamespace.nil?
-        @display_name = "#{@objnamespace}/#{@objname}"
-      else
-        @display_name = "#{@objname}"
-      end  
+      @display_name = if !@objnamespace.nil?
```
Comment (ahasunos) [inline libraries/k8sobject.rb:30]: Minor suggestion: can we change`if !@objnamespace.nil?` to `unless @objnamespace.nil?`

File: docs-chef-io/content/inspec/resources/k8s_pod.md [hunk 2]
```diff
@@ -5,59 +5,67 @@ gh_repo = "inspec"
...
-    parent = "inspec/resources/k8s"
+[menu.inspec]
+title = "k8s_pod"
+identifier = "inspec/resources/k8s/K8s Pod"
+parent = "inspec/resources/k8s"
 +++
 
 
-Use the `k8s_pod` Chef InSpec audit resource to test the configuration of...
+Use the `k8s_pod` Chef InSpec audit resource to test the configuration of specific Pods in a namespace
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pod.md:15]:
```suggestion
Use the `k8s_pod` Chef InSpec audit resource to test the configuration of specific Pod in the specified namespace
```

File: docs-chef-io/content/inspec/resources/k8s_pod.md [hunk 3]
```diff
@@ -5,59 +5,67 @@ gh_repo = "inspec"
...
 
 ## Installation
 
 ## Syntax
 
 ```ruby
-describe k8s_pod do
+describe k8s_pods(namespace: "NAMESPACE", name: "NAME") do
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pod.md:22]:
```suggestion
describe k8s_pod(namespace: "NAMESPACE", name: "NAME") do
```

File: docs-chef-io/content/inspec/resources/k8s_pod.md [hunk 4]
```diff
@@ -5,59 +5,67 @@ gh_repo = "inspec"
...
   #...
 end
 ```
 
 ## Parameters
 
-`PARAMETER`
-: PARAMETER DESCRIPTION
+`type`
+: type of the K8s resource that is for query
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pod.md:30]: As this is `k8s_pod` resource I don't think this is required here as we are setting the type to pods internally by default. @sathish-progress ?

File: docs-chef-io/content/inspec/resources/k8s_pod.md [hunk 5]
```diff
@@ -5,59 +5,67 @@ gh_repo = "inspec"
...
-: PARAMETER DESCRIPTION
+`namespace`
+: namespace of the resource
 
 ## Properties
 
-`PROPERTY`
-: PROPERTY DESCRIPTION
+`uid`
+: UID of the resource
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pod.md:38]: returns the UID of the specified Pod.

File: docs-chef-io/content/inspec/resources/k8s_pod.md [hunk 6]
```diff
@@ -5,59 +5,67 @@ gh_repo = "inspec"
...
 
-`PROPERTY`
-: PROPERTY DESCRIPTION
+`name`
+: Name of the resource
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pod.md:41]: returns the Name of the specified Pod.

File: docs-chef-io/content/inspec/resources/k8s_pod.md [hunk 7]
```diff
@@ -5,59 +5,67 @@ gh_repo = "inspec"
...
+
+`namespace`
+: Namespace of the resource
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pod.md:44]: same goes for namespace, resource_version, kine and metadata

**Review by Vasu1105 — APPROVED** [review]
Some suggestions for the docs. Also felt there should be one example with actual values. Else all look good to me.

File: docs-chef-io/content/inspec/resources/k8s_pod.md (hunk 5 above)
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pod.md:38]: UID of the Pod

File: test/unit/libraries/k8s_pod_test.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+            kind: 'pod',
+            status: {
+              phase: 'running'
+            },
+            metadata: {
+              uid: 'abcd1234',
+              name: 'pod1',
+              namespace: 'default',
+              resourceVersion: 1234,
+              annotations: [],
```
Comment (Vasu1105) [inline test/unit/libraries/k8s_pod_test.rb:19]: Annotations and Labels are k8s resource object with key value and not a array. May be we need  to have a look at this.

File: docs-chef-io/content/inspec/resources/k8s_pod.md (hunk 5 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/k8s_pod.md:38]: ok

File: docs-chef-io/content/inspec/resources/k8s_pod.md (hunk 7 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/k8s_pod.md:44]: ok

File: docs-chef-io/content/inspec/resources/k8s_pods.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+`type`
+: type of the K8s resource that is for query
+
+`namespace`
+: namespace of the resource
+
+## Properties
+
+`uid`
+: UID of the resource
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/k8s_pods.md:38]: Its Pod rite? as per the singular k8s_pod resource?

File: docs-chef-io/content/inspec/resources/k8s_pods.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+`name`
+: Name of the resource
+
+`namespace`
+: Namespace of the resource
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/k8s_pods.md:44]: here too

File: docs-chef-io/content/inspec/resources/k8s_pods.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+`resource_version`
+: resource version of the resource
+
+`kind`
+: resource type
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/k8s_pods.md:50]: here...

File: docs-chef-io/content/inspec/resources/k8s_pods.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+`metadata`
+: metadata for the resource
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/k8s_pods.md:53]: here also...

File: docs-chef-io/content/inspec/resources/k8s_pods.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+**The pods for default namespaces should exist**
+
+```ruby
+ describe k8s_pods do
+  it { should exist }
+end
+```
+
+**The pods should for given namespace exist**
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/k8s_pods.md:65]: namespaces?

**Review by dkumaras — COMMENTED** [review]
Pls check and confirm.

File: docs-chef-io/content/inspec/resources/k8s_pod.md [hunk 14]
```diff
@@ -12,57 +12,57 @@ parent = "inspec/resources/k8s"
...
+describe k8s_pod(namespace: "NAMESPACE", name: "NAME") do
   #...
 end
 ```
 
 ## Parameters
 
 `type`
-: type of the K8s resource that is for query
+: Type of the K8s resource that is for a query.
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pod.md:30]: This type property should be removed from here. This is not required.

File: docs-chef-io/content/inspec/resources/k8s_pods.md [hunk 15]
```diff
@@ -27,42 +26,42 @@ end
 ## Parameters
 
 `type`
-: type of the K8s resource that is for query
+: Type of the K8s resource that is for a query.
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pods.md:29]: Here also remove the `type` property. It's not required.

File: docs-chef-io/content/inspec/resources/k8s_pods.md [hunk 16]
```diff
@@ -27,42 +26,42 @@ end
...
+### Pods for default namespaces must exist
 
 ```ruby
  describe k8s_pods do
   it { should exist }
 end
 ```
 
-**The pods should for given namespace exist**
+### Pods for specified namespaces must exist
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pods.md:64]: namespace should be singular as it get all pods in the given namespace.

File: docs-chef-io/content/inspec/resources/k8s_pods.md [hunk 17]
```diff
@@ -27,42 +26,42 @@ end
...
+: Resource type of the Pod.
 
 `metadata`
-: metadata for the resource
+: Metadata for the Pod.
 
 ## Examples
 
-**The pods for default namespaces should exist**
+### Pods for default namespaces must exist
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_pods.md:56]: Same namespace should be singular here.

File: docs-chef-io/content/inspec/resources/k8s_pod.md (hunk 14 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/k8s_pod.md:30]: ok

**Review by dkumaras — COMMENTED** [review]
Added inputs provided

**Pull request merged**